File: src/main/java/kr/devport/api/domain/wiki/repository/WikiSectionChunkRepositoryCustom.java
package kr.devport.api.domain.wiki.repository;

import kr.devport.api.domain.wiki.entity.WikiSectionChunk;

import java.util.List;

public interface WikiSectionChunkRepositoryCustom {

    List<ScoredChunkRow> findSimilarChunksWithScore(String projectExternalId, String queryEmbedding, int limit);

    List<ScoredChunkRow> findSimilarChunksGlobalWithScore(String queryEmbedding, int limit);

    List<ScoredChunkRow> findLexicalCandidates(String projectExternalId, String question, int limit);

    record ScoredChunkRow(WikiSectionChunk chunk, double score) {
    }
}

File: wiki/repository-jpa/src/main/kotlin/kr/devport/api/domain/wiki/repository/WikiSectionChunkRepositoryCustom.kt
package kr.devport.api.domain.wiki.repository

import kr.devport.api.domain.wiki.entity.WikiSectionChunk

interface WikiSectionChunkRepositoryCustom {
    fun findSimilarChunksWithScore(
        projectExternalId: String,
        queryEmbedding: String,
        limit: Int,
    ): List<ScoredChunkRow>

    fun findSimilarChunksGlobalWithScore(
        queryEmbedding: String,
        limit: Int,
    ): List<ScoredChunkRow>

    fun findLexicalCandidates(
        projectExternalId: String,
        question: String,
        limit: Int,
    ): List<ScoredChunkRow>

    data class ScoredChunkRow(
        val chunk: WikiSectionChunk,
        val score: Double,
    )
}
